### PR TITLE
statistics: improve the memory usage for stats cache

### LINKED ISSUE #46585: improve memory usage for stats LFU cache

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #46585

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 1]
```diff
@@ -168,6 +168,12 @@ func (s *LFU) dropMemory(item *ristretto.Item) {
 	// why add before again? because the cost will be subtracted in onExit.
 	// in fact, it is after - before
 	s.addCost(after)
+	// When the memory usage of the cache exceeds the maximum value, Many item need to evict. But
+	// ristretto'c cache execute the evict operation when to write the cache. for we can evict as soon as possible,
+	// we will write some fake item to the cache. fake item have a negative key, and the value is nil.
+	if s.Cost() > s.cache.MaxCost() {
```
Comment (qw4990) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:174]: Better to add some tests for this.

Comment (hawkingrei) [reply statistics/handle/cache/internal/lfu/lfu_cache.go:174]: Done

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 2]
```diff
@@ -144,15 +145,15 @@ func (s *LFU) onEvict(item *ristretto.Item) {
 }
 
 func (s *LFU) dropMemory(item *ristretto.Item) {
+	if s.closed.Load() {
+		return
+	}
```
Comment (qw4990) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:150]: Better to keep it unchanged.

**Review by qw4990 — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

File: statistics/handle/cache/internal/lfu/lfu_cache.go (hunk 2 above)
Comment (hawkingrei) [reply statistics/handle/cache/internal/lfu/lfu_cache.go]: Done

**Review by winoros — APPROVED** [review]

**Pull request merged**